book/section3/didcomm.md

```diff
@@ -2,11 +2,92 @@
+
+## Protocols
+
+DIDComm defines not just message formats but also a framework for protocols—standardized sequences of messages that accomplish specific tasks. These protocols enable interoperability by ensuring that different implementations understand the same message sequences and semantics.
+
+The protocol framework includes mechanisms for discovering which protocols an agent supports and for versioning protocols as they evolve. This allows for graceful upgrades and backward compatibility in the ecosystem.
+
+Several common protocols have emerged in the SSI ecosystem:
+
+- **Connection establishment protocols** define how two parties can establish a secure DIDComm channel
+- **Credential issuance protocols** standardize how verifiable credentials are offered and issued
+- **Credential presentation protocols** define how credentials can be requested and presented
+- **Trust ping protocols** provide simple mechanisms to verify that a communication channel is active
+
+By standardizing these interaction patterns, DIDComm enables different implementations to work together seamlessly, fostering an open ecosystem rather than siloed solutions.
+
+You can find all currently published protocols in [here](https://didcomm.org/search/?page=1), as these are maintained by the communitity, you are also encouraged to build and define your own protocols that fit your particular use case, even if you may not decide to publish them.
+
+## Example interaction
+
+To illustrate how DIDComm works in practice, consider a typical credential issuance scenario:
+
+1. An issuer and holder establish a DIDComm connection, exchanging DIDs and service endpoints
+2. The issuer sends an offer message indicating which credentials it can provide
+3. The holder responds with a request message for a specific credential
+4. The issuer creates the credential and sends it in an issuance message
+5. The holder acknowledges receipt with an acknowledgment message
+
+Throughout this interaction, all messages are encrypted end-to-end, ensuring that even if they pass through intermediaries, the content remains private between the issuer and holder. The messages might travel over different transport mechanisms—perhaps starting with an HTTPS connection but switching to Bluetooth for later messages if the parties come into proximity.
+
+In mediated scenarios, the flow becomes more complex but even more powerful. A holder might receive messages through a mediator service that provides consistent message delivery even when the holder's device is offline. The mediator receives encrypted messages, stores them until the holder connects, and then forwards them without ever being able to read the encrypted contents.
+
+## Benefits
+
+DIDComm delivers several crucial benefits that advance the core principles of Self-Sovereign Identity:
```

Comment:
Here's an example of where it's fine to relate DIDComm to SSI.  This is ok because it's not inferring that SSI requires DIDComm, just that DIDComm helps support the principals of SSI